[stable/sonarqube] upgrades LTS to latest patch

usmonster opened the pull request:
Also updates ingress apiVersion, adding support for later k8s versions.
(Modernizes documentaiton accordingly.)

Signed-off-by: Usman Akeju <akeju00+github@gmail.com>

#### Is this a new chart
> NOTE: We're experiencing a high volume of PRs to this repo and reviews will be delayed. Please host your own chart repository and submit your repository to the Helm Hub instead of this repo to make them discoverable to the community. [Here](https://github.com/helm/hub/blob/master/Repositories.md) is how to submit new chart repositories to the Helm Hub.

Nope!

#### What this PR does / why we need it:

It upgrades the stable SonarQube chart to refer to the latest patch of its LTS version.
It also makes the chart installable on Kubernetes >= 1.16. (see also #19023)

#### Which issue this PR fixes

  - fixes #19652, fixes #19723, fixes #18999

#### Special notes for your reviewer:

I had to add the `apiVersion` update in order to test it.
As such, this pull request can probably supersede #19023.

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

lachie83 commented on stable/sonarqube/templates/ingress.yaml:
```diff
@@ -1,7 +1,7 @@
 {{- $serviceName := include "sonarqube.fullname" . -}}
 {{- $servicePort := .Values.service.externalPort -}}
-apiVersion: extensions/v1beta1
+apiVersion: networking.k8s.io/v1
```
How about this to make it compatible with versions <= k8s 1.13?

```suggestion
{{ if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" }}
apiVersion: networking.k8s.io/v1beta1
{{ else }}
apiVersion: extensions/v1beta1
{{ end -}}
```

usmonster replied:
~I don't think it's required, since this was apparently available since v1.8 (unless I'm misreading [the changelog](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.14.md#deprecations)).~

~Can you confirm it's a problem on k8s versions <= 1.13 as-is?~

Oops! Sorry, I read too fast—I was thinking of `NetworkPolicy`. Applying the changes. 👍

Thanks for your quick feedback!

lachie83 commented on REVIEW_GUIDELINES.md:
````diff
@@ -279,7 +279,7 @@ ingress:
 ```yaml
 {{- if .Values.ingress.enabled -}}
-apiVersion: extensions/v1beta1
+apiVersion: networking.k8s.io/v1
````
How about this to make it compatible with versions <= k8s 1.13

```suggestion
{{ if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" }}
apiVersion: networking.k8s.io/v1beta1
{{ else }}
apiVersion: extensions/v1beta1
{{ end -}}
```

usmonster replied:
I wonder if it would be fine to exclusively recommend `networking.k8s.io/v1beta1` in the guidelines to avoid the added complexity of the conditional? Chart owners can decide whether or not to support older versions of k8s, but maybe we shouldn't mention deprecated API versions in the guidelines? (Just a philosophical question, no strong opinion here.)

usmonster replied:
Given that a bunch of folks are still on older versions of k8s, I take back my suggestion. Once old versions become _really_ old, then I'm sure documentation will be updated. (I'll let you resolve the conversation when you are OK with it, @lachie83.)

usmonster committed:
[stable/sonarqube] support also k8s <= v1.13

Signed-off-by: Usman Akeju <akeju00+github@gmail.com>

usmonster committed:
[stable/sonarqube] upgrades LTS to latest patch

Also updates ingress apiVersion, adding support for later k8s versions.
(Modernizes documentaiton accordingly.)

Signed-off-by: Usman Akeju <akeju00+github@gmail.com>
